### PR TITLE
#367 Initial re-work of CSV file support.

### PR DESCRIPTION
**Pull request by knausj85**
@rntz Please test.

### CODE REVIEW

File: code/user_settings.py [hunk 1]
```diff
@@ -13,37 +11,33 @@
...
-    path = SETTINGS_DIR / file_name
+def get_list_from_csv(
+    filename: str, headers: Tuple[str, str], default: Dict[str, str] = {}
+):
+    """Retrieves list from CSV"""
+    path = SETTINGS_DIR / filename
+    assert filename.endswith(".csv")
 
-    # Create the file if it doesn't exist
-    if not SETTINGS_DIR.is_dir():
```
Comment (rntz) [inline code/user_settings.py:28]: Is there a reason this code to auto-create the settings directory got dropped? Seems like we'll need this for first-time users.

File: code/user_settings.py [hunk 2]
```diff
@@ -13,37 +11,33 @@
...
+    # ability to reload this script for us when the resource changes
+    with resource.open(str(path), "r") as f:
+        rows = list(csv.reader(f))
 
+    # print(str(rows))
     mapping = {}
     if len(rows) >= 2:
         actual_headers = rows[0]
         if not actual_headers == list(headers):
             print(
```
Comment (rntz) [inline code/user_settings.py:45]: Might want to use logging.warning here instead of just printing.

File: code/user_settings.py [hunk 3]
```diff
@@ -56,95 +50,11 @@ def _load_csv_dict(
                 output, spoken_form = row[:2]
                 if len(row) > 2:
                     print(
```
Comment (rntz) [inline code/user_settings.py:58]: Ditto re logging.warning.

**Review by rntz — COMMENTED** [review]
Haven't actually tested yet, just a read-through.

File: code/user_settings.py (hunk 1 above)
Comment (knausj85) [reply code/user_settings.py:28]: It's there:
https://github.com/knausj85/knausj_talon/blob/50f55ef7abb829421cc494fae0a8bdeb3688f8df/code/user_settings.py#L11

File: code/user_settings.py (hunk 3 above)
Comment (knausj85) [reply code/user_settings.py:58]: if this is useful for you I would vote to write up an issue,  and we can address this more broadly.   I generally have very little output,  so haven't bothered

**Comment by rntz** [comment]
did some basic testing, seemed to work, I should probably try testing on a clean install though

**Comment by knausj85** [comment]
A clean install seems a bit overkill- simply deleting the settings folder while talon isn't running would probably suffice

**Pull request merged**